bodhi/metadata.py

```diff
@@ -170,181 +180,104 @@ def _fetch_updates(self):
-        })
-
-        self._insert(root, 'id', text=update.alias)
-        self._insert(root, 'title', text=update.title)
-        self._insert(root, 'release', text=update.release.long_name)
-        self._insert(root, 'issued', attrs={
-            'date': update.date_pushed.strftime('%Y-%m-%d %H:%M:%S'),
-        })
+        rec = cr.UpdateRecord()
+        rec.version = __version__
+        rec.fromstr = config.get('bodhi_email')
+        rec.status = update.status.value
+        rec.type = update.type.value
+        rec.id = to_bytes(update.alias)
+        rec.title = to_bytes(update.title)
+        rec.summary = to_bytes('%s %s update' % (update.get_title(),
+                                                 update.type.value))
+        rec.description = to_bytes(update.notes)
+        rec.release = to_bytes(update.release.long_name)
+        rec.rights = config.get('updateinfo_rights')
+
+        rec.issued_date = update.date_pushed
         if update.date_modified:
-            self._insert(root, 'updated', attrs={
-                'date': update.date_modified.strftime('%Y-%m-%d %H:%M:%S'),
-            })
+            rec.updated_date = update.date_modified
 
-        # Build the references
-        refs = self.doc.createElement('references')
-        for cve in update.cves:
-            self._insert(refs, 'reference', attrs={
-                'type': 'cve',
-                'href': cve.url,
-                'id': cve.cve_id
-            })
-        for bug in update.bugs:
-            self._insert(refs, 'reference', attrs={
-                'type': 'bugzilla',
-                'href': bug.url,
-                'id': bug.bug_id,
-                'title': bug.title
-            })
-        root.appendChild(refs)
-
-        # Update description
-        self._insert(root, 'description', text=update.notes)
-
-        # The package list
-        pkglist = self.doc.createElement('pkglist')
-        collection = self.doc.createElement('collection')
-        collection.setAttribute('short', update.release.name)
-        self._insert(collection, 'name', text=update.release.long_name)
+        col = cr.UpdateCollection()
+        col.name = to_bytes(update.release.long_name)
+        col.shortname = to_bytes(update.release.name)
 
         for build in update.builds:
-            kojiBuild = None
             try:
                 kojiBuild = self.builds[build.nvr]
             except:
                 kojiBuild = self.koji.getBuild(build.nvr)
+
             rpms = self.koji.listBuildRPMs(kojiBuild['id'])
             for rpm in rpms:
-                filename = "%s.%s.rpm" % (rpm['nvr'], rpm['arch'])
+                pkg = cr.UpdateCollectionPackage()
+                pkg.name = rpm['name']
+                pkg.version = rpm['version']
+                pkg.release = rpm['release']
+                pkg.epoch = rpm['epoch'] or '0'
+                pkg.arch = rpm['arch']
+
+                # TODO: how do we handle UpdateSuggestion.logout, etc?
```

Comment:
Right now the metadata for this looks like `<reboot_suggested>True</reboot_suggested>`.

Sounds like we need a more flexible schema, maybe something like `<suggest type="logout"/>`